core/java/lib/src/main/java/com/microsoft/kiota/core/HttpCore.java

```diff
@@ -84,6 +71,67 @@ public HttpCore(@Nonnull final AuthenticationProvider authenticationProvider, @N
             }
         });
     }
+    @Nonnull
+    public <ModelType> CompletableFuture<ModelType> sendPrimitiveAsync(@Nonnull final RequestInfo requestInfo, @Nonnull final Class<ModelType> targetClass, @Nullable final ResponseHandler responseHandler) {
+        return addBearerIfNotPresent(requestInfo).thenCompose(x -> {
+            final HttpCoreCallbackFutureWrapper wrapper = new HttpCoreCallbackFutureWrapper();
+            this.client.newCall(getRequestFromRequestInfo(requestInfo)).enqueue(wrapper);
+            return wrapper.future;
+        }).thenCompose(response -> {
+            if(responseHandler == null) {
+                final ResponseBody body = response.body();
+                try {
+                    if(targetClass == Void.class) {
+                        return CompletableFuture.completedStage(null);
+                    } else {
+                        if(targetClass == InputStream.class) {
+                            return CompletableFuture.completedStage((ModelType)body.byteStream());
+                        }
+                        final String rawJson = body.string();
+                        final JsonParseNode rootNode = new JsonParseNode(rawJson);
+                        if(targetClass == Boolean.class) {
```

Comment:
consider extracting method here as well.

Reply:
refactored to about duplicating the casting.